### PR TITLE
Add exclude and include functions as wrappers around cut_region syntax

### PR DESCRIPTION
**Pull request by rcjackson**
## PR Summary
This PR provides wrapper functions around the cut_region routines for filtering data. The current syntax in Yt is awkward and this functionality provides a cleaner way of creating cut_regions.

## PR Checklist

- [x] Code passes flake8 checker
- [x] New features are documented, with docstrings and narrative docs
- [x] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

File: doc/source/analyzing/filtering.rst [hunk 1]
```diff
@@ -86,7 +86,34 @@ other data object to generate images, examine its values, etc.
...
+   import yt
+   ds = yt.load('Enzo_64/DD0042/data0042')
+   ad = ds.all_data()
+   overpressure_and_fast = ds.include_above('pressure', 1e-14)
+   # You can chain include_xx and exclude_xx to produce the intersection of cut regions
+   overpressure_and_fast = overpressure_and_fast.include_above('velocity_magnitude', 1e2, 'km/s')
+
+   print('Density of all data: ad["density"] = \n%s' % ad['density'])
+   print('Density of "overpressure and fast" data: ad["density"][overpressure_and_fast] = \n%s' %
+          ad['density'][overpressure_and_fast])
```
Comment (brittonsmith) [inline doc/source/analyzing/filtering.rst]: This syntax (`ad['density'][overpressure_and_fast]`) is a bit different from the conventional syntax that looks something like
```
overpressure_and_fast['density']
```
Are both allowed? If so, I would suggest making a specific point to mention this.

File: doc/source/analyzing/filtering.rst [hunk 2]
```diff
@@ -86,7 +86,34 @@ other data object to generate images, examine its values, etc.
...
+
+The following exclude and include functions are supported:
```
Comment (brittonsmith) [inline doc/source/analyzing/filtering.rst]: Might want to also add mentions to `<include/exclude<_<above/below>`.

File: yt/data_objects/data_containers.py [hunk 3]
```diff
@@ -1785,7 +1785,332 @@ def cut_region(self, field_cuts, field_parameters=None):
...
+        field : string
+            The field in which the conditional will be applied.
+        value : float
+            The minimum value that will not be masked in the output
+            YTCutRegion.
+        units : string or None
+            The units of the value threshold. None will use the default units
+            given in the field.
+
+        Returns
```
Comment (brittonsmith) [inline yt/data_objects/data_containers.py]: Could you add an Example for each one of these functions?  This is likely going to be the place that most users come for documentation on how to use this.

File: yt/data_objects/data_containers.py [hunk 4]
```diff
@@ -1785,7 +1785,332 @@ def cut_region(self, field_cuts, field_parameters=None):
...
+            field_cuts = ('~np.isnan(obj["' + field + '"])')
+        else:
+            field_cuts = ('~np.isnan(obj["' + field + '"].in_units("' + units + 
+                '"))')
+        cr = self.cut_region(field_cuts)
+        return cr
+
+    def include_below(self, field, value, units=None):
+        """
+        This function will return a YTCutRegion where onlythe regions
```
Comment (cphyc) [inline yt/data_objects/data_containers.py]: There is a small nit here.

File: doc/source/analyzing/filtering.rst [hunk 5]
```diff
@@ -86,7 +86,34 @@ other data object to generate images, examine its values, etc.
...
+wrapper functions exist that allow the user to use a simplified syntax for
+filtering out unwanted regions. Such wrapper functions are methods of
+:func: ``YTSelectionContainer3D``.
+
+.. notebook-cell::
+
+   import yt
+   ds = yt.load('Enzo_64/DD0042/data0042')
+   ad = ds.all_data()
+   overpressure_and_fast = ds.include_above('pressure', 1e-14)
```
[the remark this answers is not in the document]
Comment (rcjackson) [reply doc/source/analyzing/filtering.rst]: I fixed this with my latest commit.

File: doc/source/analyzing/filtering.rst (hunk 1 above)
Comment (rcjackson) [reply doc/source/analyzing/filtering.rst]: It's not allowed. I went ahead and fixed this part of the documentation to fix the syntax to be correct.

File: doc/source/analyzing/filtering.rst (hunk 2 above)
Comment (rcjackson) [reply doc/source/analyzing/filtering.rst]: Done and done!

File: yt/data_objects/data_containers.py (hunk 3 above)
Comment (rcjackson) [reply yt/data_objects/data_containers.py]: I added an example for each of the added functions in the latest commit.

File: doc/source/analyzing/filtering.rst [hunk 6]
```diff
@@ -69,11 +69,11 @@ used if you simply need to access the NumPy arrays:
 
     import yt
     ds = yt.load('Enzo_64/DD0042/data0042')
-    ad = ds.all_data()
+    ad = ad.all_data()
```
Comment (brittonsmith) [inline doc/source/analyzing/filtering.rst]: I think you changed the wrong `ds` to `ad`. This one should still be `ds`.

File: doc/source/analyzing/filtering.rst [hunk 7]
```diff
@@ -86,7 +86,39 @@ other data object to generate images, examine its values, etc.
...
+wrapper functions exist that allow the user to use a simplified syntax for
+filtering out unwanted regions. Such wrapper functions are methods of
+:func: ``YTSelectionContainer3D``.
+
+.. notebook-cell::
+
+   import yt
+   ds = yt.load('Enzo_64/DD0042/data0042')
+   ad = ds.all_data()
+   overpressure_and_fast = ds.include_above('pressure', 1e-14)
```
Comment (brittonsmith) [inline doc/source/analyzing/filtering.rst]: This `ds` should be `ad`.

**Comment by brittonsmith** [comment]
Just one minor nit, but otherwise, this is great!

File: doc/source/analyzing/filtering.rst (hunk 7 above)
Comment (chummels) [reply doc/source/analyzing/filtering.rst]: agreed.

**Comment by ngoldbaum** [comment]
@yt-fido test this please

**Comment by ngoldbaum** [comment]
Opening and closing to re-trigger appveyor which is stuck

**Comment by munkm** [comment]
This is such a fantastic feature @rcjackson! Thank you so much for your contribution.

**Pull request merged**